### PR TITLE
Implements LifecycleNode constructor enableCommunicationInterface

### LINKED ISSUE #799: LifecycleNode constructor missing  enableCommunicationInterface parameter added in ROS2 Galactic
This issue is a reminder that LifecycleNode constructor needs to be updated to reflect the additional constructor parameter `enableCommunicationInterface`. This parameter was added in the Galactic rcl release and has a default = true for backwards compatibility. We committed a quick partial implementation of the default behavior for the 0.19 release.  Now we need to enable the ability to disable lifecycle service interfaces.

https://docs.ros.org/en/foxy/Releases/Release-Galactic-Geochelone.html#rcl-lifecycle-and-rclcpp-lifecycle

### PR DESCRIPTION
**Pull request by wayneparrott**
lifecycle-node.js
lifecycle-node.d.ts
rcl_lifecycle_binding.cpp
main.ts
* added support for enableCommunicationInterface param

index.js
* added support for enableCommunicationInterface param

index.d.ts
* added support for enableCommunicationInterface param
* added @deprecated tag to createNode() & createLifecycleNode()

test-lifecycle.js
* added 2 test to cover pos and neg versions of
enableCommunicationInterface parameter

Fix #799

### CODE REVIEW

**Comment by wayneparrott** [comment]
will fix the silly EOL prettier issues asap and update.

**Comment by minggangw** [comment]
Thanks for making this progress!

File: src/rcl_lifecycle_bindings.cpp [hunk 1]
```diff
@@ -53,6 +53,7 @@ static v8::Local<v8::Object> wrapTransition(
 NAN_METHOD(CreateLifecycleStateMachine) {
   RclHandle* node_handle = RclHandle::Unwrap<RclHandle>(
       Nan::To<v8::Object>(info[0]).ToLocalChecked());
+  bool enableCommunicationInterface = Nan::To<bool>(info[1]).FromJust();
```
Comment (minggangw) [inline src/rcl_lifecycle_bindings.cpp]: nit: enableCommunicationInterface => enable_communication_interface

File: src/rcl_lifecycle_bindings.cpp [hunk 2]
```diff
@@ -62,30 +63,30 @@ NAN_METHOD(CreateLifecycleStateMachine) {
...
-
+  const rosidl_service_type_support_t* cs =
+      GetServiceTypeSupport("lifecycle_msgs", "ChangeState");
+  const rosidl_service_type_support_t* gs =
+      GetServiceTypeSupport("lifecycle_msgs", "GetState");
 #if ROS_VERSION >= 2105
+
   rcl_lifecycle_state_machine_options_t options =
       rcl_lifecycle_get_default_state_machine_options();
+  options.enable_com_interface = enableCommunicationInterface;
```
Comment (minggangw) [inline src/rcl_lifecycle_bindings.cpp]: ditto

**Review by minggangw — APPROVED** [review]
LGTM with some nits, thanks for the PR!

File: src/rcl_lifecycle_bindings.cpp (hunk 1 above)
Comment (wayneparrott) [reply src/rcl_lifecycle_bindings.cpp]: I renamed the var to `enable_com_interface` which is the same name as the options property.

**Pull request merged**